### ISSUE TITLE
Making `_text` available to other secondary UUID columns

### ISSUE THREAD

**Issue by vpratfr** [spatie/laravel-binary-uuid#45]
Currently, the trait offers some goodies on the main uuid field (the PK). Models may have foreign keys to other models according to their UUID. It would be nice to open some of the features on those attributes too.

### The _text computed attribute

other UUID attributes should also have a corresponding computed string representation. e.g. `relation_uuid_text` would return string version of `relation_uuid`

This could be handled in the same way as dates are handled. Have a `protected $uuids = [ 'relation_uuid' ]` on the model to let the trait know which attributes are handled as UUIDs.

Would you be ok for a PR about that?

**Comment by brendt** [comment]
I think saving the UUID both in textual and binary form is a good idea. I've been thinking about how I'd implement this package if we'd start over again (read: v2).

I think the primary key field should just be the textual UUID, with an extra indexed field added to store the binary data. This binary field could be used for high performance searches, relations, etc. but other things like route binding wouldn't need to worry about converting UUIDs.

Feel free to share your thoughts.

**Comment by vpratfr** [comment]
One big disadvantage of UUIDs over regular (big) integers (with or without auto-increment) is the storage overhead. 

By adding both versions of the UUID within the DB, you clearly make that disadvantage really become a concern.

I personnally do not even expose the UUIDs to the world. I have a hashed version of it (base62) which feels better in the URL : 21 characters instead of 36. I have a trait which handles automatic encoding/decoding of that UUID into what I call the `human_id`. 

My current approach is very simple: I override the getAttribute method and check the `$key` parameter. If it ends with `_human_id`, then I automatically return the Base62 encoded version of the corresponding `_uuid` attribute.

Here is the code for the function:

```
    // Automatic computed attributes from corresponding uuid attribute:
    // e.g. $this->my_relation_human_id => encoded version of $this->my_relation_uuid

    public function getAttribute($key)
    {
        if (Str::endsWith($key, '_human_id'))
        {
            $uuidKey = \str_replace('_human_id', '_uuid', $key);
            $value = parent::getAttribute($uuidKey);
            return $value!==null ? self::uuidToHumanId(static::decodeUuid($value)) : null;
        }

        return parent::getAttribute($key);
    }
```

You could do exactly the same thing: everything which ends with `_uuid_text` will equal to the string version of the corresponding _uuid attribute.

**Comment by vpratfr** [comment]
Note: this is an automated way of providing it.

If you prefer a more user-controlled way, then instead of detecting if `$key` ends with `_uuid_text`, then we could simply check if the key belongs to the model defined array `$this->uuids` (just like it is done by the Model class for date attributes that need to be casted to Carbon instances.

**Comment by isocroft** [comment]
This issue has been fixed by #49 . So it's safe to close ?
